lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -949,26 +953,34 @@ public function AliasIdentificationVariable()
     }
 
     /**
-     * AbstractSchemaName ::= identifier
+     * AbstractSchemaName ::= qualified_name | aliased_name | identifier
+     *
+     * @param bool $validateName Whether to validate that the parsed name refers to an existing class.
      *
      * @return string
      */
-    public function AbstractSchemaName()
+    public function AbstractSchemaName($validateName = true)
```

Comment:
Validating that the class exists seems to be a separate concern to me. Why did you introduce it?